### ISSUE TITLE
Method and field with same name not detected as error

### ISSUE THREAD

**Issue by jeffreymorlan** [Microsoft/TypeScript#13141]
**TypeScript Version:**  2.0.5, 2.1.4, master

**Code**

```ts
class C {
	a(): number { return 0; }
	a: number;
}
```

**Expected behavior:**

Since you can't have a method and a field with the same name, there should be a "Duplicate identifier 'a'" error, as there was in TS 1.8, and as there still is if the field comes first.

**Actual behavior:**

No error. The field takes precedence (`.a` property has type of `number`, is not callable).

**Comment by errorx666** [comment]
Methods are defined on the prototype object, while properties are defined on the instances themselves, so it is technically possible for both to exist, though the own property will shadow the prototype property.
I would expect an error here as well.

**Comment by aluanhaddad** [comment]
@errorx666 I think you are right. There seems to be something strange going on here because while the property is indeed seen as shadowing the method, intellisense is behaving oddly

![image](https://cloud.githubusercontent.com/assets/5034656/21454098/aeecb920-c8e3-11e6-8b98-ecc06fe9c4d0.png)
![image](https://cloud.githubusercontent.com/assets/5034656/21454108/bc3ff43e-c8e3-11e6-8703-8db12fc5c7c9.png)
